test_util/cluster.py

```diff
@@ -372,6 +368,11 @@ def wait_for_mesos_metric(cluster, host, key, value):
             add_config_path=add_config_path,
             rexray_config_preset=cluster.rexray_config_preset,
         )
+
+        # Generate node upgrade script
+        output = installer.generate_node_upgrade_script()
+        upgrade_script_path = output.decode('utf-8').strip('\n').split("Node upgrade script URL: ", 1)[1]
```

Comment:
See my note above about splitting by line to get the upgrade script URL --  if the URL were for some reason in the middle of the output here, the `upgrade_script_path` would include a bunch of unrelated text and the test wouldn't fail until we attempt to curl. If the URL isn't in the right place or the right format, we should fail immediately rather than waiting for the curl.